safenode/src/domain/dbc_genesis.rs

```diff
@@ -43,25 +54,156 @@ pub enum Error {
     FailedToParseReason(#[from] DbcError),
 }
 
-/// Create the genesis DBC.
-/// The genesis DBC is the first DBC in the network. It is created without
-/// a source transaction, as there was nothing before it.
-#[allow(clippy::result_large_err)]
-pub fn create_genesis() -> GenesisResult<Dbc> {
-    let secret_key = bls::SecretKey::from_hex(GENESIS_DBC_SK).map_err(sn_dbc::Error::Blsttc)?;
+lazy_static! {
+    /// Load the genesis DBC.
+    /// The genesis DBC is the first DBC in the network. It is created without
+    /// a source transaction, as there was nothing before it.
+    pub static ref GENESIS_DBC: Dbc = match sn_dbc::Dbc::from_hex(GENESIS_DBC_HEX) {
+        Ok(dbc) => dbc,
+        Err(err) => panic!("Failed to read genesis DBC: {err:?}"),
+    };
+}
+
+/// Return if provided DbcTransaction is genesis parent tx.
+pub(crate) fn is_genesis_parent_tx(parent_tx: &DbcTransaction) -> bool {
+    parent_tx == &GENESIS_DBC.src_tx
+}
+
+/// Returns a dbc with the requested number of tokens, for use by E2E test instances.
+pub async fn get_tokens_from_faucet(amount: Token, to: PublicAddress, client: &Client) -> Dbc {
+    send(load_faucet_wallet(client).await, amount, to, client).await
+}
+
+pub(crate) async fn send(
+    from: LocalWallet,
+    amount: Token,
+    to: PublicAddress,
+    client: &Client,
+) -> Dbc {
+    if amount.as_nano() == 0 {
+        panic!("Amount must be more than zero.");
+    }
+
+    let mut wallet_client = WalletClient::new(client.clone(), from);
+    let new_dbc = wallet_client
+        .send(amount, to)
+        .await
+        .expect("Tokens shall be successfully sent.");
+
+    let mut wallet = wallet_client.into_wallet();
+    wallet
+        .store()
+        .await
+        .expect("Wallet shall be successfully stored.");
+    wallet
+        .store_created_dbc(new_dbc.clone())
+        .await
+        .expect("Created dbc shall be successfully stored.");
+
+    new_dbc
+}
+
+/// Load or create faucet wallet.
+pub async fn load_faucet_wallet(client: &Client) -> LocalWallet {
+    let genesis_wallet = load_genesis_wallet().await;
+
+    println!("Loading faucet...");
+    let mut faucet_wallet = create_faucet_wallet().await;
+
+    use super::wallet::{DepositWallet, Wallet};
+    let faucet_balance = faucet_wallet.balance();
+    if faucet_balance.as_nano() > 0 {
+        println!("Faucet wallet balance: {faucet_balance}");
+        return faucet_wallet;
+    }
+
+    // Transfer to faucet. We will transfer almost all of the genesis wallet's
+    // balance to the faucet, only leaving enough to pay for transfer fees.
+
+    let initial_fee_margin = 500_000;
```

Comment:
This will be fixed later, so that we actually transfer everything from genesis.

Reply:
(And there will be solution to the live transfers from first faucet so that there is no issue with transfer fees.)